packages/auth/src/signature/middleware.test.ts

```diff
@@ -401,6 +401,101 @@ describe('Signature Service', (): void => {
       scope.isDone()
     })
 
+    test('middleware succeeds if BYPASS_SIGNATURE_VALIDATION is true with bad signature', async (): Promise<void> => {
+      const defaultByPassSignatureValidation = Config.bypassSignatureValidation
+      Config.bypassSignatureValidation = true
+
+      const scope = nock(KEY_REGISTRY_ORIGIN)
+        .get(keyPath)
+        .reply(200, {
+          jwk: testClientKey.jwk,
+          client: TEST_CLIENT
+        } as ClientKey)
+
+      const ctx = await createContextWithSigHeaders(
+        {
+          headers: {
+            Accept: 'application/json'
+          },
+          url: '/',
+          method: 'POST'
+        },
+        {},
+        {
+          client: {
+            display: TEST_CLIENT_DISPLAY,
+            key: {
+              proof: 'httpsig',
+              jwk: testClientKey.jwk
+            }
+          }
+        },
+        privateKey,
+        deps
+      )
+
+      ctx.headers['signature'] = 'wrong-signature'
+
+      await grantInitiationHttpsigMiddleware(ctx, next)
+
+      expect(ctx.response.status).toEqual(200)
+      expect(next).toHaveBeenCalled()
+
+      Config.bypassSignatureValidation = defaultByPassSignatureValidation
+
+      scope.isDone()
+    })
+
+    test('middleware fails if BYPASS_SIGNATURE_VALIDATION is false with bad signature', async (): Promise<void> => {
```

Comment:
I think you could remove this test, and just add a check that `bypassSignatureValidation` is false in:
https://github.com/interledger/rafiki/blob/c8d42f60386d1c5a0e387c4e8bd30d6f74846f41/packages/auth/src/signature/middleware.test.ts#L372

Reply:
Agree, I had added it because the `Config` is a global and I wasn't convinced it was resetting between runs. 

Your suggestions below look better and solve this so I can do both and get rid of the extra test.